makedef.pl

```diff
@@ -145,6 +145,9 @@ BEGIN
     elsif ($define{HAS_SETLOCALE}) {
         $define{USE_LOCALE} = 1;
     }
+    else {
+        $define{NO_LOCALE} = 1;
```

Comment:
What is with the `USE_LOCALE` vs `NO_LOCALE` identifiers in this `.pl` or in the whole build process? What if both are defined by the end user on the command line, or 1 automatically, and 1 by the user?

`makedef.pl` has 2 existing conditionals for `USE_LOCALE`, and none for `NO_LOCALE`.  `makedef.pl` ignores whatever `NO_LOCALE` means, and only cares about `&& $define{USE_LOCALE}` test.

Why suddenly make `NO_LOCALE` defined and `== 1` instead of force undef-ing `USE_LOCALE` or `croak()`ing?